components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java

```diff
@@ -418,17 +435,42 @@ public String getRelativeInternalURL() {
             return relativeInternalUrl;
         }
 
-        private String fetchAbsoluteUrl() throws URLBuilderException {
+        private String fetchAbsolutePublicUrl() throws URLBuilderException {
+
+            StringBuilder absolutePublicUrl = new StringBuilder();
+            if (StringUtils.isBlank(protocol)) {
+                throw new URLBuilderException("Protocol of service URL is not available");
```

Comment:
Add a fullstop at the end. Check other places as well.

Reply:
Fixed https://github.com/wso2/carbon-identity-framework/pull/3111/commits/beb53f7154b1bbf61aeb6b1130af4fc849b4b681